Documentation/ExtensionArchitecture/HowTo/ExtendExtbaseModel/Index.rst

```diff
@@ -0,0 +1,208 @@
+
+For example if the model is found in :file:`Classes/Domain/Model/SomeModel.php`
+the repository is located in
+:file:`Classes/Domain/Repository/SomeModelRepository.php`.
+
+*   If you do not find this repository but found the model the extension might
+    not use an Extbase repository. This tutorial does not help you in this
+    case as it can only be applied to Extbase repositories.
+*   If you find a repository in this name-schme but it does not extend
+    directly or indirectly the class
+    :php:`TYPO3\CMS\Extbase\Persistence\Repository` you are also not dealing
+    with an Extbase repository.
+*   If the repository is final it cannot be extended.
+
+In all these cases refer to the extensions documentation on how to extend it.
+
+..  _extending-extbase-model_extend_original_model:
+
+Extend the original model
+-------------------------
+
+We assume you already extended the SQL table and TCA (table configuration array)
+as described in :ref:`Extending TCA <extending-tca>`. Extend the
+original model by a custom class in your extension:
+
+..  literalinclude:: _MyExtendedModel.php
+    :language: php
+    :caption: EXT:my_extension/Classes/Domain/Model/MyExtendedModel.php
+
+Add all additional fields that you require. By convention the database
+fields are usually prefixed with your extensions name and so are the
+names in the model.
+
+..  _extending-extbase-model_register_extended_model:
+
+Register the extended model
+---------------------------
+
+The extended model needs to be registered for Extbase persistence in file
+:file:`Configuration/Extbase/Persistence/Classes.php`.
+
+..  literalinclude:: _Classes.php
+    :language: php
+    :caption: EXT:my_extension/Configuration/Extbase/Persistence/Classes.php
+
+
+..  _extending-extbase-model_extend_original_repository:
+
+Extend the original repository
+------------------------------
+
+Likewise extend the original repository:
+
+..  literalinclude:: _MyExtendedModelRepository.php
+    :language: php
+    :caption: EXT:my_extension/Classes/Domain/Repository/MyExtendedModelRepository.php
+
+The rule that a repository must follow the name schema of the model also
+applies when extending model and repository. So the new repositorys name must
+end on "Repository" and it must be in the directory :file:`Domain/Repository`.
+
+If you have no need for additional repository methods you can leave the body of
+this class empty. However for Extbase internal reasons you have to create this
+repository even if you need no additional functionality.
+
+
+..  _extending-extbase-model_register_extended_repository:
+
+Register the extended repository
+--------------------------------
+
+The extended repository needs to be registered with Extbase in your extensions
+:file:`EXT:my_extension/ext_localconf.php`. This step tells
+Extbase to use your repository instead of the original one whenever the original
+repository is requested via Dependency Injection in a controller or service.
```

Comment:
Is this really DI or just GU:makeInstance()? AFAIK XCLASS only works with GU:mI().

Reply:
So in the controller I will usually request the repository via Dependency Injection? In some cases also via GU:makeInstance?

Reply:
Don't used XCLASSes in the last years. If it works for you with DI, I am fine with it.

Reply:
For me it works with DI, though I hate doing this :-(